cli/command/service/rollback.go

```diff
@@ -0,0 +1,67 @@
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/versions"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+func newRollbackCommand(dockerCli *command.DockerCli) *cobra.Command {
+	options := newServiceOptions()
+
+	cmd := &cobra.Command{
+		Use:     "rollback [OPTIONS]",
+		Aliases: []string{"rollback"},
+		Short:   "Rollback a service",
+		Args:    cli.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runRollback(dockerCli, options, args[0])
+		},
+	}
+
+	return cmd
+}
+
+func runRollback(dockerCli *command.DockerCli, options *serviceOptions, serviceID string) error {
+	apiClient := dockerCli.Client()
+	ctx := context.Background()
+
+	service, _, err := apiClient.ServiceInspectWithRaw(ctx, serviceID, types.ServiceInspectOptions{})
+	if err != nil {
+		return err
+	}
+
+	updateOpts := types.ServiceUpdateOptions{}
+	spec := &service.Spec
+
+	if versions.LessThan(dockerCli.Client().ClientVersion(), "1.28") {
```

Comment:
I think we should just hide this command on earlier versions, so we don't need to check version here. You can do that by setting `Tags` on the `cobra.Command` above

```go
Tags: map[string]string{"version": "1.31"}
```

That way we can always set `updateOpts.Rollback` to `"previous"`